module_utils/certificate_lsr/providers/certmonger.py

```diff
@@ -254,6 +256,21 @@ def request_certificate(self):
         command = [getcert_bin]
 
         if self.exists_in_certmonger:
+            # if certificate exists in certmonger and key-file is missing,
+            # reissuing the certificate will hang certmonger.
+            # See: https://issues.redhat.com/browse/RHEL-69043
+            keyfile = self._certmonger_metadata.get("key-file")
+            if keyfile:
+                try:
+                    _fstat = os.stat(keyfile)
```

Comment:
Instead of `os.stat` with exception handling, you can do something like

```python
            if keyfile and not os.path.exists(keyfile):
                    self.module.fail_json(
                        "Resubmiting a request without the private key "
                        "file may hang certmonger. Please, stop monitoring "
                        "certificate '{0}' before reissuing.".format(
                            self._certmonger_metadata.get('nickname')
                        )
                    )
````

Reply:
> I'm not sure I shouldn't process the stat result.
> 
> What if the file has been removed and a directory created?

ok, but in with the current code, `_fstat` is not used, and CodeQL flags it

Reply:
I change to use `os.path.isfile`. More than that and we would be looking into the file content, which I believe is way out of scope here.